parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -886,6 +951,15 @@ private static void serializeColumnIndexes(
     }
   }
 
+  private int toIntWithCheck(long size) {
```

Comment:
In other places we use `java.util.Math.toIntExact(long)`.

Reply:
Yes you are right, I was thinking of keeping it this way here(like in ColumnChunkPageWriteStore ) so the message can be clearer to the caller of what failed

Reply:
It's fine for me.
Then, I would suggest using a similar approach of implementation then they have it in `toIntExact(long)`. The one you used is not universal (think of negative numbers out of int).